code/modules/antagonists/heretic/items/hunter_rifle.dm

```diff
@@ -134,6 +133,35 @@
 	damage = 30
 	stamina = 30
 	projectile_phasing =  PASSTABLE | PASSGLASS | PASSGRILLE | PASSCLOSEDTURF | PASSMACHINE | PASSSTRUCTURE | PASSDOORS
+	///The mob that is currently inside the bullet
+	var/mob/stored_mob
+
+/obj/projectile/bullet/strilka310/lionhunter/fire(angle, atom/direct_target)
+	. = ..()
+	if(isliving(firer))
+		var/mob/living/living_firer = firer
+		if(IS_HERETIC(living_firer))
+			living_firer.forceMove(src)
+			stored_mob = living_firer
+
+/obj/projectile/bullet/strilka310/lionhunter/Destroy()
+	if(stored_mob)
+		stored_mob.forceMove(loc)
+	return ..()
```

Comment:
No side effects in destroy: This should be in on hit, impact, `on_range`.

Though, I am certain projectiles do not handle this cleanly at all, so we should probably keep this here, with a stack trace:
```suggestion
/obj/projectile/bullet/strilka310/lionhunter/Destroy()
	if(stored_mob)
		stack_trace("Lionhunter bullet qdel'd with its firer still inside!")
		stored_mob.forceMove(loc)
	return ..()
```

Reply:
Also, the reference is unhandled. You will want to `null` the `stored_mob` var. 
You can do it easily with `Exited`
```dm
/obj/projectile/bullet/strilka310/lionhunter/Exited(atom/movable/gone, direction)
	. = ..()
	if(gone == stored_mob)
		stored_mob = null
```